R/map_rd2markdown.R

```diff
@@ -12,14 +12,11 @@
 map_rd2markdown <- function(frags, ..., collapse = NULL) {
   out <- lapply(frags, rd2markdown, ...)
   out <- Filter(Negate(is.null), out)
-  out <- Filter(function(i) {
-    tag <- attr(i, "tag")
-    if (!is.null(tag) && tag == "cr") {
-      nchar(i)
-    } else {
-      nchar(trimws(i))
-    }
-  }, out)
-  if (!is.null(collapse)) out <- paste0(out, collapse = collapse)
+  out <- Filter(nchar, out)
+  if (!is.null(collapse)) {
+    out <- paste0(out, collapse = collapse)
+    # Irrelevant form the rendering point of view, just for styling reasons
+    out <- trim_extra_newlines(out)
+  } 
```

Comment:
I believe that this issue fixes the underlying problem and hopefully most of the edge cases. For some reason we had that `trimws()` call in the filter that made us drop alone `\n` that otherwise would produce proper paragraphs. The reason was that `\n\n` signs are splitted into separate TEXT tags and in some cases were dropped because of that filter. Now it should all work just fine! (fingers crossed, thumbs hold) @dgkf

Reply:
I do think we want to trim most newlines. Using `Rd2txt` as a reference, lines re-flowed. It is only when we have two consecutive `\n` that we want to introduce a paragraph break, similar to markdown. 

```tex
\\name{test}
\\title{test}
\\description{
  one 
  two
}
```

Will render as

```
test

Description:

     one two

```

Reply:
Frankly, I don't see what's wrong with this example, it looks correct. Also, I don't see how a change to that filter function affects that. The change (keep in mind change is a little bit above, here I marked it a bit incorrectly) makes us simply take all signs that were already present in the rd file, we don't introduce new ones. Or am I missing something?

Reply:
Now I see what you mean, so this should be fine:

```
## test

one
two
```

because it still renders as:

> ## test
>
> one two

---

Unfortunately, I think there are still cases where this will be too assumptive. For example, an Rd can contain trailing whitespace before a newline, which would be considered a newline in markdown but is wrapped with `Rd2txt`:

```tex
% using "•" to indicate spaces
\\name{test}
\\title{test}
\\description{
••one••
••two••
}
```

```
test

Description:

     one two
     
```

but if we just kept all the whitespace as-is, it would display in markdown as

> ## test
> 
> one  
> two

Reply:
I'm sorry but I still don't see the problem in your example. I've prepared a dummy Rd like the one above (had to manually put spaces as roxygen removes them) and run it through the current version of `rd2markdown` from this branch. This is the raw result

```
"\n\n# test\n\none \n two \n\n"
```

So we have indeed one new line sign between one and two. But this is only one. Pandoc will ignore it while rendering. However, I tried I couldn't recreate the output you've provided using Pandoc. Obviously, it will work differently in gitlab, as it's using a pseudo-markdown where every new line is rendered as is and I don't think we should account for that.

Would you mind sharing how you created that markdown display at the end of the previous message?